src/__tests__/e2e.test.js

```diff
@@ -442,4 +442,57 @@ describe('e2e', () => {
 
     expect(global.console.error).toHaveBeenCalledTimes(1);
   });
+
+  it('will dispatch different data if props changed', () => {
+    @track(props => ({ data: props.data }))
+    class Top extends React.Component {
+      render() {
+        return this.props.children;
+      }
+    }
+
+    @track({ page: 'Page' })
+    class Page extends React.Component {
+      @track({ event: 'buttonClick' })
+      handleClick = jest.fn();
+      render() {
+        return <span onClick={this.handleClick}>Click Me</span>;
```

Comment:
Linting is failing on this line, you can just `// eslint-disable-line jsx-a11y/no-static-element-interactions`